File: academic-system-java/src/main/java/org/example/academic/system/repository/TxtUserRepository.java
package org.example.academic.system.repository;

import org.example.academic.system.model.Role;
import org.example.academic.system.model.User;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

public class TxtUserRepository implements UserRepository {

    private final List<User> users = new ArrayList<>();

    public TxtUserRepository() {
        try (BufferedReader reader = new BufferedReader(new FileReader("users.txt"))) {
            String line;
            while ((line = reader.readLine()) != null) {
                String[] parts = line.split(",");
                if (parts.length == 3) {
                    users.add(new User(parts[0].trim(), parts[1].trim(), Role.valueOf(parts[2].trim())));
                }
            }
        } catch (IOException e) {
            users.add(new User("admin", "admin", Role.ADMIN));
            users.add(new User("professor", "professor", Role.PROFESSOR));
        }
    }

    @Override
    public Optional<User> findByUsername(String username) {
        return users.stream()
                .filter(user -> user.getUsername().equalsIgnoreCase(username))
                .findFirst();
    }
}

File: academic-system-python/academic_system/repository/txt_user_repository.py
import os
from typing import Optional

from .user_repository import UserRepository
from ..model.role import Role
from ..model.user import User


class TxtUserRepository(UserRepository):
    FILE_NAME = "users.txt"

    def __init__(self):
        self._users: dict[str, User] = self._load_users()

    def _load_users(self) -> dict:
        users = {}
        if not os.path.exists(self.FILE_NAME):
            users["admin"] = User("admin", "admin", Role.ADMIN)
            users["professor"] = User("professor", "professor", Role.PROFESSOR)
            return users
        try:
            with open(self.FILE_NAME, "r", encoding="utf-8") as f:
                for line in f:
                    line = line.strip()
                    if not line:
                        continue
                    parts = line.split(",")
                    if len(parts) == 3:
                        username, password, role_str = parts
                        role = Role[role_str.strip().upper()]
                        users[username.strip()] = User(username.strip(), password.strip(), role)
        except Exception as e:
            print(f"Erro ao carregar usuarios: {e}")
        return users

    def find_by_username(self, username: str) -> Optional[User]:
        return self._users.get(username)
